storage/blob.go

```diff
@@ -368,41 +368,46 @@ var (
 // pagination token and other response details.
 //
 // See https://msdn.microsoft.com/en-us/library/azure/dd179352.aspx
-func (b BlobStorageClient) ListContainers(params ListContainersParameters) (ContainerListResponse, error) {
+func (b BlobStorageClient) ListContainers(params ListContainersParameters) (clr ContainerListResponse, err error) {
 	q := mergeParams(params.getParameters(), url.Values{"comp": {"list"}})
 	uri := b.client.getEndpoint(blobServiceName, "", q)
 	headers := b.client.getStandardHeaders()
 
-	var out ContainerListResponse
 	resp, err := b.client.exec("GET", uri, headers, nil)
 	if err != nil {
-		return out, err
+		return clr, err
 	}
-	defer resp.body.Close()
+	defer func() {
+		err = resp.body.Close()
+	}()
 
-	err = xmlUnmarshal(resp.body, &out)
-	return out, err
+	err = xmlUnmarshal(resp.body, &clr)
+	return clr, err
 }
 
 // CreateContainer creates a blob container within the storage account
 // with given name and access level. Returns error if container already exists.
 //
 // See https://msdn.microsoft.com/en-us/library/azure/dd179468.aspx
-func (b BlobStorageClient) CreateContainer(name string, access ContainerAccessType) error {
+func (b BlobStorageClient) CreateContainer(name string, access ContainerAccessType) (err error) {
 	resp, err := b.createContainer(name, access)
 	if err != nil {
 		return err
 	}
-	defer resp.body.Close()
+	defer func() {
+		err = resp.body.Close()
+	}()
 	return checkRespCode(resp.statusCode, []int{http.StatusCreated})
 }
 
 // CreateContainerIfNotExists creates a blob container if it does not exist. Returns
 // true if container is newly created or false if container already exists.
-func (b BlobStorageClient) CreateContainerIfNotExists(name string, access ContainerAccessType) (bool, error) {
+func (b BlobStorageClient) CreateContainerIfNotExists(name string, access ContainerAccessType) (notExists bool, err error) {
```

Comment:
I named the bool because if I name the error, all returning stuff should be named.
The error is named so errors from `defer` can be returned and checked.